cli/mfg.c

```diff
@@ -786,6 +788,191 @@ static int security_config_set(int argc, char **argv)
 	return 0;
 }
 
+#if HAVE_LIBCRYPTO
+static int load_sig_from_file(FILE *sig_fimg, uint8_t *sig)
+{
+	ssize_t rlen;
+
+	rlen = fread(sig, 1, SWITCHTEC_SIG_LEN, sig_fimg);
+
+	if (rlen < SWITCHTEC_SIG_LEN)
+		return -1;
+
+	return 0;
+}
+
+static int kmsk_already_exist(uint8_t *kmsk,
```

Comment:
Seems like this could be in the library. switchtec_security_state_has_key()... I'd put the state as the first argument.

Reply:
This function is only used for this command so is kept here for now. A little hesitant to add it to API, which means one more (probably unnecessary) function to support.

Reply:
All the functions are only used for that command.... Looks like a useful function to me to support into the future.... Better than having users open code it themselves because then you have to support the data structure for ever into the future which is much worse.

Reply:
Make sense. I will add it to API function.